src/main/java/com/minecolonies/coremod/colony/jobs/JobDeliveryman.java

```diff
@@ -172,13 +195,13 @@ public void addRequest(@NotNull final IToken token)
      */
     public void finishRequest(@NotNull final boolean successful)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "successful" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)